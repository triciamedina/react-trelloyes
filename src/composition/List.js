import React from 'react';
import './List.css';
import Card from './Card'

function List(props) {
    return (
        <section className='List'>
            <header className='List-header'>
                <h2>{props.header}</h2>
                {/* added new random card button*/}
                <button 
                    className='.List-add-button'
                    // event listener calls  add random card function 
                    // and passes the id of the list as an argument
                    onClick={() => props.onAddRandomCard(props.listId)}
                >
                    add random card
                </button>
            </header>
            <div className='List-cards'>
                {props.cards.map(item => 
                    <Card 
                        key={item.id}
                        // added a prop for the card id 
                        cardId={item.id}
                        title={item.title}
                        content={item.content} 
                        onDeleteCard={props.onDeleteCard}
                    />
                )}
            </div>
        </section>
    )
}

export default List;