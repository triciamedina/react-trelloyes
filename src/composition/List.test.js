import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('List component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        const cardArray = [{id: "0", title: "test card", content: "lorem ipsum"}];
        ReactDOM.render(<List key={0} header='test' cards={cardArray}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('renders the UI as expected', () => {
        const cardArray = [{id: "0", title: "test card", content: "lorem ipsum"}];
        const tree = renderer
            .create(<List key={0} header='test' cards={cardArray} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
