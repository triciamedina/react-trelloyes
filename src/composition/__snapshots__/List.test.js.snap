// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      test
    </h2>
    <button
      className=".List-add-button"
      onClick={[Function]}
    >
      add random card
    </button>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3>
        test card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
